### PR TITLE
SQL Plugin

### PR DESCRIPTION
**Pull request by SferaDev**
- Add SQL plugin to `XataClient`
- Add Kysely dialect for Xata

### CODE REVIEW

**Comment by richardgill** [comment]
Would it be possible to add a description to this PR just explaining what the goal is. I also noticed the build is failing, is that expected?

File: .changeset/great-tigers-scream.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@xata.io/client': patch
```
Comment (SferaDev) [inline .changeset/great-tigers-scream.md:2]:
```suggestion
'@xata.io/client': minor
```

File: .changeset/khaki-eagles-complain.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@xata.io/kysely': patch
```
Comment (SferaDev) [inline .changeset/khaki-eagles-complain.md:2]:
```suggestion
'@xata.io/kysely': minor
```

File: packages/client/src/sql/parameters.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+      result += '\\\\x' + val[i].toString('hex');
+    } else {
+      result += escapeElement(prepareValue(val[i]));
+    }
+  }
+  result = result + '}';
+  return result;
+}
+
+// Code distilled from pg/lib/utils.js
```
Comment (richardgill) [inline packages/client/src/sql/parameters.ts:30]: Would it be possible to link to this code? Does this code come with tests (or could we write our own), it feels very testable / edge-casey.

File: packages/plugin-client-kysely/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# @xata.io/kysely
+
+A [Kysely](https://github.com/kysely-org/kysely) dialect for [Xata](https://xata.io), using the [Xata serverless driver for TypeScript](https://github.com/xataio/client-ts).
```
Comment (richardgill) [inline packages/plugin-client-kysely/README.md:3]: cc: @joan-ing 

This shows up inside npm, do we want to link to our docs. 

What is going on with the kysely docs, who will be writing them?

**Review by richardgill — APPROVED** [review]
Left some comments, but these changes look good to me.

File: packages/client/src/sql/parameters.ts (hunk 3 above)
Comment (SferaDev) [reply packages/client/src/sql/parameters.ts:30]: Other libraries are using the same helper, also it's not a direct copy paste of a function, instead it tries to mimick the logic within that file.

File: packages/plugin-client-kysely/README.md (hunk 4 above)
Comment (SferaDev) [reply packages/plugin-client-kysely/README.md:3]: We will have our own docs, this readme will be for direct links from Kysely

File: test/integration/sql.test.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+beforeEach(async (ctx) => {
+  await hooks.beforeEach(ctx);
+});
+
+afterEach(async (ctx) => {
+  await hooks.afterEach(ctx);
+});
+
+describe('SQL proxy', () => {
```
Comment (SferaDev) [inline test/integration/sql.test.ts:29]:
```suggestion
describe.skip('SQL proxy', () => {
```
